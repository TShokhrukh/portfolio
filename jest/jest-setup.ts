import '@testing-library/jest-dom'
import { TextEncoder, TextDecoder } from 'util'
import { IntersectionObserverMock } from './mock/IntersectionObserverMock '

// @ts-ignore
global.TextEncoder = TextEncoder
// @ts-ignore
global.TextDecoder = TextDecoder
// @ts-ignore
global.IntersectionObserver = IntersectionObserverMock
