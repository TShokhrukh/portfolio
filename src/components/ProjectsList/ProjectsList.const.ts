import { cn } from '@bem-react/classname'

export const cnProjectsList = cn('ProjectsList')
