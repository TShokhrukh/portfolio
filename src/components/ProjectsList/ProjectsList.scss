.ProjectsList {
  --projects-list-grid-template-columns: 1;
  --list-gap: #{$spacer * 2};
  grid-template-columns: var(--projects-list-grid-template-columns);
}

@include media-breakpoint-up(xl) {
  .ProjectsList {
    --projects-list-grid-template-columns: repeat(4, 1fr);
  }
}