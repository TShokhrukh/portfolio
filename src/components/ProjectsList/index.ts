export * from './build'
export * from './__types__'
export { ProjectsList as ProjectsListBase } from './ProjectsList'
