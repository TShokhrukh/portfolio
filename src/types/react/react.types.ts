import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
