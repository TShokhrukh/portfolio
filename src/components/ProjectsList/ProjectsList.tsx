import React, { FC } from 'react'
import { cnProjectsList } from './ProjectsList.const'
import { ProjectCard } from '~/components/ProjectCard/build'
import { List } from '~/components/List/List'
import type { IProjectsListProps } from './__types__/ProjectsList.types'
import './ProjectsList.scss'

export const ProjectsList: FC<IProjectsListProps> = ({
  projects,
  className,
  ...props
}) => {
  return <List {...props}
    className={cnProjectsList({}, [className])}
    items={projects} itemComponent={ProjectCard} itemPropKey='project' />
}
