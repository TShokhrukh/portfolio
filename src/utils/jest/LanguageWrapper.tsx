import React, { ReactNode } from 'react'
import { TranslationsProvider } from '@eo-locale/react'

import type { Locale } from '@eo-locale/core'

interface IWrapperProps {
  children: ReactNode,
  language?: string,
  locales?: Locale[]
}

const defLocales: Locale[] = [
  {
    language: 'en',
    messages: {}
  }
]

export const LanguageWrapper: React.FC<IWrapperProps> = ({
  children,
  language = 'en',
  locales = defLocales
}) => {
  return (
    <TranslationsProvider language={language} locales={locales} onError={() => null}>
      {children}
    </TranslationsProvider>
  )
}
