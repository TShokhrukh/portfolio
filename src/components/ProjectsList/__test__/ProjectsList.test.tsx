/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { LanguageWrapper } from '~/utils/jest/LanguageWrapper'
import { ProjectsList } from '../ProjectsList'
import type { IProject } from '../__types__'

describe('ProjectsList', () => {
  test('must return the component template', () => {
    const projects: IProject[] = [
      {
        id: '1',
        href: 'http://localhost:9000/project_card/1',
        desc: { en: 'project_card 1' },
        image: 'http://localhost:9000/image.png',
        name: 'project name 1',
        stack: ['JavaScript']
      },
      {
        id: '2',
        href: 'http://localhost:9000/project_card/2',
        desc: { en: 'project_card 2' },
        image: 'http://localhost:9000/image.png',
        name: 'project name 2',
        stack: ['TypeScript', 'JavaScript']
      }
    ]
    render(
      <LanguageWrapper>
        <ProjectsList projects={projects} data-testid='projects_list' />
      </LanguageWrapper>
    )

    expect(screen.getByTestId('projects_list')).toMatchSnapshot()
  })
})
