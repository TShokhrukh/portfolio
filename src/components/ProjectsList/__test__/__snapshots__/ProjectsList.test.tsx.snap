// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsList must return the component template 1`] = `
<div
  class="List ProjectsList"
  data-testid="projects_list"
>
  <a
    class="Link Card ProjectCard"
    href="http://localhost:9000/project_card/1"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p
      class="Typography Typography_color_white"
    >
      project name 1
    </p>
    <p
      class="Typography ProjectCard-Desc"
    >
      project_card 1
    </p>
    <div
      class="ProjectCard-Stacks"
    >
      <span
        class="Badge Badge_color_purple"
      >
        JavaScript
      </span>
    </div>
    <img
      alt="project name 1"
      class="Img ProjectCard-Image"
    />
  </a>
  <a
    class="Link Card ProjectCard"
    href="http://localhost:9000/project_card/2"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p
      class="Typography Typography_color_white"
    >
      project name 2
    </p>
    <p
      class="Typography ProjectCard-Desc"
    >
      project_card 2
    </p>
    <div
      class="ProjectCard-Stacks"
    >
      <span
        class="Badge Badge_color_purple"
      >
        TypeScript
      </span>
      <span
        class="Badge Badge_color_purple"
      >
        JavaScript
      </span>
    </div>
    <img
      alt="project name 2"
      class="Img ProjectCard-Image"
    />
  </a>
</div>
`;
